Spatial_oM/ShapeProfiles/TaperFlangeChannelProfile.cs

```diff
@@ -0,0 +1,97 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Geometry;
+using System.Collections.ObjectModel;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using BH.oM.Quantities.Attributes;
+
+namespace BH.oM.Spatial.ShapeProfiles
+{
+    [Description("C-shaped profile with tapered flanges of the same length and thickness.")]
+    public class TaperFlangeChannelProfile : BHoMObject, IProfile, IImmutable
+    {
+
+        /***************************************************/
+        /**** Properties                                ****/
+        /***************************************************/
+        public virtual ShapeType Shape { get; } = ShapeType.Channel;
+
+        [Length]
+        [Description("Full depth between the extreme fibres of the flanges.")]
+        public virtual double Height { get; }
+
+        [Length]
+        [Description("Full width between the extreme fibres of the web to the toe of the flange.")]
+        public virtual double FlangeWidth { get; }
+
+        [Length]
+        public virtual double WebThickness { get; }
+
+        [Length]
+        [Description("Mean thickness of the flange, taken at 1/2 the flange width")]
+        public virtual double FlangeThickness { get; }
+
+        [Angle]
+        [Description("slope of the flange inner faces, generally around 0.16 radians.")]
```

Comment:
Small suggestion of stating slightly larger range of commonly used values
```suggestion
        [Description("Slope of the flange inner faces, generally around 0.05 to 0.17 radians.")]
```